feature: double scan

Fehniix opened the pull request:
- Added double scan mode
- Added custom picker option view

Fehniix committed:
feature: updated ui dl 4feb

- WIP
- Abstracted views requiring a header/footer view in own view controller
- Abstracted header bar in own view controller/xib pair

ludovicoIBM committed:
feature: update ui dl4feb

- bugfix in set content view in HFBackViewController

Fehniix committed:
feature: updated ui dl 4feb

- WIP: added footer view

eapuzzo committed:
feature: updated ui dl 4feb

- Fix UI WIP

Fehniix committed:
feature: updated ui dl 4feb

- Updated UI constraints to match requested requirements

Fehniix committed:
Merge branch 'develop' into feature/updated-ui-dl-4feb

# Conflicts:
#	Verifier.xcodeproj/project.pbxproj

Fehniix committed:
feat: updated ui dl 4feb

- Updated localizable strings as per requirements

eapuzzo committed:
feature: update ui dl 4feb

- Update icon_logo image

eapuzzo committed:
feature: update ui dl feb4

- Add circle info image
- Fix verificationNeeded result view link

eapuzzo committed:
Feature: updated ui dl 4feb

- Fix Home title
- Changed switch camera icon
- Add info circle button
- Change from AppLabel to TextView in AppAlertViewVontroller
- Updated info text view string in localizable

eapuzzo committed:
feature: update ui dl 4feb

- Fix alert condition

Fehniix committed:
feat: updated ui dl 4feb

- Added CSS styles to HTML attributed string in localized strings

Fehniix committed:
feat: updated ui dl 4feb

- Differentiated scan mode info alert from scan mode unset alert

Fehniix committed:
feat: updated ui dl 4feb

- Fixed home title label font

Fehniix committed:
feat: updated ui dl 4feb

- Updated switch camera icon size
- Updated color of custom alert

Fehniix committed:
feat: custom picker

- WIP: added first custom picker implementation

Fehniix committed:
feat: updated ui dl 4feb

- Added missing localizable strings
- Increased size of info button to 30px
- Added missing itEntry case in header bar

Fehniix committed:
feat: updated ui dl 4feb

- Updated info icon padding

Fehniix committed:
feat: updated ui dl 4feb

- Updated localizable strings

Fehniix committed:
feat: updated ui dl 4feb

- Implemented a "extendable button", a UIButton whose "tappable" area can be widened across both axes
- Widened info button tappable area

Light2288 committed:
feat: new ui

change version text style
fix version string
remove duplicate string
change faq link
change version

Fehniix committed:
feat: custom picker

- WIP: interactive generic picker implementation

Fehniix committed:
feature: custom picker

- Fully functional scrollable options with dynamic height based on content size

Fehniix committed:
feature: custom picker

- Reassigned tap responsibility to parent controller rather than to each individual custom picker option

Fehniix committed:
feature: custom picker

- Fully functional non-polymorphic custom picker

Fehniix committed:
feature: custom picker

- Minor UI improvements

Fehniix committed:
Merge branch 'feature/updated-ui-dl-4feb' into feature/custom-picker

# Conflicts:
#	Verifier.xcodeproj/project.pbxproj

Fehniix committed:
Merge branch 'feature/updated-ui-dl-4feb' into feature/custom-picker

Fehniix committed:
feat: custom picker

- Adapted UI as per latest request, with corresponding logic

eapuzzo committed:
feature: custom picker

- Fix Localizable strings
- Add alert condition for missing settings

eapuzzo committed:
feature: custom picker

- Add new Constanst for faq and verification alert settings
- Update link logic (use Settings)
- Removed old link logic from Link+Values and Status+Configuration

eapuzzo committed:
feature: custom picker

- Fix Settings name

Fehniix committed:
Merge branch 'develop' into feature/custom-picker

# Conflicts:
#	DGCAVerifier/Models/Internal/Link.swift
#	DGCAVerifier/Pages/Home/HomeViewController.swift
#	DGCAVerifier/SupportingFiles/en.lproj/Localizable.strings
#	DGCAVerifier/SupportingFiles/it.lproj/Localizable.strings
#	DGCAVerifier/Utils/ModelExtensions/Link+Values.swift
#	DGCAVerifier/Utils/ModelExtensions/Status+Configuration.swift
#	Verifier.xcodeproj/project.pbxproj

Fehniix committed:
feat: custom picker

- Fixed bug addressing iOS versions < 13.0 with required constraints not being available to be changed at runtime

eapuzzo committed:
feature: custom picker

- Update PopUp setting string

eapuzzo committed:
feature: custom picker

- Fix alert message

eapuzzo committed:
feature: double scan

- Add Status Singleton WIP

Fehniix committed:
feat: double scan

- WIP: conditional animated presentation

Light2288 committed:
feat: new scan mode selection menu

ui fixes

eapuzzo committed:
Merge branch 'feature/custom-picker' into feature/double-scan

Light2288 committed:
feat: new scan mode selector

fix pop ups

Fehniix committed:
Merge branch 'feature/custom-picker' into feature/double-scan

eapuzzo committed:
feature: double scan

- WIP Add tickView

eapuzzo committed:
Merge branch 'feature/double-scan' of https://github.com/ministero-salute/it-dgc-verificaC19-ios into feature/double-scan

Fehniix committed:
feat: double scan

- Implemented TestOnlyValidator along with associated ValidatorBuilder changes
- Minor refactor
- Minor changes
- WIP: add validator hook to UI

Fehniix committed:
Merge branch 'feature/double-scan' of github.com:ministero-salute/it-dgc-verificaC19-ios into feature/double-scan

eapuzzo committed:
feature: double scan

- WIP
- Add new Verification page
- AppButton change constraint to label to self centerY
- Update ResultView and Verification view priority

Fehniix committed:
feat: double scan

- Added verbose debug logs
- Implemented state logic to allow for double scan paths
- Added showCamera to VerificationCoordinator

Fehniix committed:
feat: double scan

- Moved constants to localizable strings
- Fixed no test available button style

Fehniix committed:
feat: double scan

- Fixed custom picker option description view dynamic height

astagi reviewed (changes requested):
Please do not use tabs

Fehniix commented:
> Please do not use tabs

Sounds good. 

Could you point me to files that require indentation adjustments? Thank you.

Fehniix committed:
feat: double scan

- Improved UI implementation of double scan view
- Added constraint print adapter for third-party tools

Fehniix committed:
feat: double scan

- Converted tabs to spaces

Fehniix committed:
feat: double scan

- Updated mocked settings

Fehniix commented:
@astagi files were updated as per request.

astagi commented:
@Fehniix I see a lot of files using tabs instead of spaces, resulting in an indentation mess e.g. DGCAVerifier/BusinessRules/Internal/Validators/DGCValidatorBuilder.swift . This is impossible to review and approve, see changes and look what’s wrong.

Fehniix committed:
feat: double scan

- Converted all tabs to whitespaces

Fehniix commented:
@astagi I see, agreed. I have replaced all tabs within the project with whitespaces. We will set our IDEs to automatically indent using whitespaces rather than tabs. Let me know if there is anything else I can do - thank you!

astagi commented:
> @astagi I see, agreed. I have replaced all tabs within the project with whitespaces. We will set our IDEs to automatically indent using whitespaces rather than tabs. Let me know if there is anything else I can do - thank you!

thank you @Fehniix !
